monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/db/manager/JDBCDatabaseManager.java

```diff
@@ -257,6 +258,31 @@ public String getMgtApiUrl(String groupId, String nodeId) {
         return mgtApiUrl;
     }
 
+    @Override
+    public String getAccessToken(String groupId, String nodeId) {
+        String query = "SELECT ACCESS_TOKEN FROM HEARTBEAT WHERE GROUP_ID=? AND NODE_ID=?;";
+        Connection con = null;
+        PreparedStatement statement = null;
+        String accessToken = "";
+        try {
+            con = getConnection();
+            statement = con.prepareStatement(query);
+            statement.setString(1, groupId);
+            statement.setString(2, nodeId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                accessToken = resultSet.getString(1);
+            }
+        } catch (SQLException e) {
+            throw new DashboardServerException("Error occurred while retrieving access token of node: " + nodeId
+                                               + " in group " + groupId, e);
+        } finally {
+            closeStatement(statement);
```

Comment:
Can we use a try-with-resources statement here?

Reply:
and for other similar cases

Reply:
This is fixed with https://github.com/wso2/product-mi-tooling/pull/197